src/components/percentage/percentage.test.jsx

```diff
@@ -0,0 +1,261 @@
+      expect(screen.getByLabelText('the label')).toHaveClass('test-class');
+    });
+  });
+
+  describe('cssContainer API', () => {
+    it('sets container className', () => {
+      render(<Percentage {...requiredProps} cssContainer="test-class" />);
+      expect(screen.getByLabelText('the label').parentElement.parentElement).toHaveClass('test-class');
+    });
+  });
+
+  describe('dirty API', () => {
+    it('is unset without any changes', () => {
+      const ref = createRef();
+      render(<Percentage {...requiredProps} ref={ref} />);
+      expect(ref.current.dirty).toBe(false);
+    });
+
+    it('is set with any changes', () => {
+      const ref = createRef();
+      render(<Percentage {...requiredProps} ref={ref} />);
+      userEvent.type(screen.getByLabelText('the label'), '1');
+      expect(ref.current.dirty).toBe(true);
+    });
+
+    it('is unset with changes back to initial state', () => {
+      const ref = createRef();
+      render(<Percentage {...requiredProps} ref={ref} />);
+      userEvent.type(screen.getByLabelText('the label'), '1');
+      userEvent.type(screen.getByLabelText('the label'), '{backspace}');
+      expect(ref.current.dirty).toBe(false);
+    });
+
+    it('is unset with changes and new value prop', () => {
+      const ref = createRef();
+      const { rerender } = render(<Percentage {...requiredProps} ref={ref} />);
+      userEvent.type(screen.getByLabelText('the label'), '1');
+      rerender(<Percentage {...requiredProps} ref={ref} value={1} />);
+      expect(ref.current.dirty).toBe(false);
+    });
+  });
+
+  describe('id API', () => {
+    it('sets the ID', () => {
+      render(<Percentage {...requiredProps} id="test-id" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('id', 'test-id');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Percentage {...requiredProps} id="unique-id" ref={ref} />);
+      expect(ref.current.id).toBe('unique-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('can be set', () => {
+      render(<Percentage {...requiredProps} label="Unique label" />);
+      expect(screen.getByLabelText('Unique label')).toBeInTheDocument();
+    });
+  });
+
+  describe('name API', () => {
+    it('sets the name attribute', () => {
+      render(<Percentage {...requiredProps} name="test-name" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('name', 'test-name');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Percentage {...requiredProps} name="unique-name" ref={ref} />);
+      expect(ref.current.name).toBe('unique-name');
+    });
+  });
+
+  describe('onBlur API', () => {
+    it('sets the onblur handler', () => {
+      const onBlurSpy = jest.fn();
+      render(<Percentage {...requiredProps} onBlur={onBlurSpy} />);
+      fireEvent.blur(screen.getByLabelText('the label'));
```

Comment:
We prefer to use user-events for interacting with the component. 

```suggestion
      userEvent.tab({ shift: true });
```